From 5dc4c89b6935fcfe327b77681362319e6fd407e0 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Thu, 1 Sep 2011 11:08:04 -0400
Subject: [PATCH] exportfs: do not warn if /etc/exports.d/ does not exist

It isn't uncommon for /etc/exports.d/ to not exist, and imo, it's not
that big of a deal as often times, a simple /etc/exports is sufficient.
So silently skip the case where the dir is missing.

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 utils/exportfs/exportfs.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index 986a272..7432a65 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -529,9 +529,12 @@ export_d_read(const char *dname)
 
 
 	n = scandir(dname, &namelist, NULL, versionsort);
-	if (n < 0)
+	if (n < 0) {
+		if (errno == ENOENT)
+			/* Silently return */
+			return;
 		xlog(L_NOTICE, "scandir %s: %s", dname, strerror(errno));
-	else if (n == 0)
+	} else if (n == 0)
 		return;
 
 	for (i = 0; i < n; i++) {
-- 
1.7.6

